python/ray/autoscaler/_private/aws/node_provider.py

```diff
@@ -507,13 +508,26 @@ def fillout_available_node_types_resources(
             for instance in instances_list
         }
         available_node_types = cluster_config["available_node_types"]
+        head_node_type = cluster_config["head_node_type"]
         for node_type in available_node_types:
             instance_type = available_node_types[node_type]["node_config"][
                 "InstanceType"]
             if instance_type in instances_dict:
                 cpus = instances_dict[instance_type]["VCpuInfo"][
                     "DefaultVCpus"]
+
                 autodetected_resources = {"CPU": cpus}
+                if node_type != head_node_type:
```

Comment:
We only need to autodetect the worker node type memory resources. Because the head node type memory is also can be updated from runtime. And also the head node type memory is `total_memory * (1 - REDIS_PROPORTION - OBJECT_STORE_PROPORTION)`, however the worker node type is `total_memory * (1  - OBJECT_STORE_PROPORTION)`.